modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1090,12 +1087,40 @@ def _read(cls, filepath_or_buffer, **kwargs):
             newline, quotechar = cls.compute_newline(
                 fio, encoding, kwargs.get("quotechar", '"')
             )
+
+            if nrows := kwargs["nrows"] if not should_handle_skiprows else None:
+                appprox_nlines = nrows
+            else:
+                # Estimating an approximate number of lines in the file by reading
+                # the first 10 lines, calculating the average line length in bytes
+                # and dividing the file length by the average line length.
+                encoding = encoding or "UTF-8"
+                newline_len = len(newline) if newline else 1
+                lines_len = 0
+                i = 0
+                while True:
+                    line = fio.readline()
+                    if line is None:
+                        break
+                    if len(line) == 0:
+                        continue
+                    lines_len += len(line.encode(encoding)) + newline_len
+                    if (i := i + 1) == 10:
+                        break
+                if i == 0:
+                    appprox_nlines = 0
+                else:
+                    avg_line_len = lines_len // i
+                    appprox_nlines = cls.file_size(f) // avg_line_len
+            num_partitions = max(1, (appprox_nlines * len(column_names)) // 64_000)
+            num_partitions = min(NPartitions.get(), num_partitions)
```

Comment:
may we move this heuristic out to a function and place it somewhere close to [the 'chunking' logic](https://github.com/modin-project/modin/blob/885e6ea450f0430af2f5e2409c15eeda7d5b687b/modin/core/storage_formats/pandas/utils.py#L26) so we can later reuse it in other places?